apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/edit/components/WhoToSendCard.tsx

```diff
@@ -86,121 +133,206 @@ export default function WhoToSendCard({ localSurvey, setLocalSurvey, attributeCl
+                      mainText="This action resets all filters in this survey."
+                      declineBtnLabel="Cancel"
+                      onDecline={() => {
+                        setResetAllFiltersModalOpen(false);
+                      }}
+                      confirmBtnLabel="Ok"
+                      onConfirm={() => {
+                        const updatedUserSegment = structuredClone(userSegment);
+                        if (updatedUserSegment?.filters) {
+                          updatedUserSegment.filters = [];
+                        }
+
+                        setUserSegment(updatedUserSegment);
+                        setResetAllFiltersModalOpen(false);
+                      }}
+                    />
+                  </>
+                </div>
               ) : (
-                <FunnelIcon className="mr-4 h-6 w-6 text-slate-600" />
+                <div className="flex flex-col gap-4 rounded-lg p-2">
+                  <div className="mb-2 flex items-center gap-6">
+                    <UserGroupIcon className="h-6 w-6 text-slate-700" />
+                    <div className="flex flex-col">
+                      <h3 className="font-medium text-slate-900">{localSurvey.userSegment?.title}</h3>
+                      <p className="text-sm text-slate-500">{localSurvey.userSegment?.description}</p>
+                    </div>
+                  </div>
+
+                  <div className="flex items-center gap-4">
+                    <div className="flex items-center space-x-2">
+                      <Button
+                        variant="secondary"
+                        size="sm"
+                        onClick={() => {
+                          setSegmentEditorViewOnly(!segmentEditorViewOnly);
+                        }}>
+                        {segmentEditorViewOnly ? "Hide" : "View"} Filters
+                      </Button>
+                    </div>
+
+                    {isSegmentUsedInOtherSurveys && (
+                      <Button
+                        variant="secondary"
+                        size="sm"
+                        className="flex items-center gap-2"
+                        onClick={() => handleCloneSegment()}>
+                        <p className="text-sm">Clone segment</p>
+                      </Button>
+                    )}
+                    <Button
+                      variant="secondary"
+                      size="sm"
+                      className="flex items-center gap-2"
+                      onClick={() => {
+                        if (isSegmentUsedInOtherSurveys) {
+                          setSegmentUsedModalOpen(true);
+                        } else {
+                          setIsSegmentEditorOpen(true);
+                          setSegmentEditorViewOnly(false);
+                        }
+                      }}>
+                      <p className="text-sm">Edit Segment</p>
```

Comment:
@jobenjada I find the edit Segment flow a bit confusing. 

We first create/build a segment within the editor:
<img width="750" alt="Screenshot 2024-01-24 at 12 19 45" src="https://github.com/formbricks/formbricks/assets/675065/a0be7413-8960-43f2-b17e-e0714713227d">

This is the view after a segment was saved:
<img width="759" alt="Screenshot 2024-01-24 at 12 18 45" src="https://github.com/formbricks/formbricks/assets/675065/0ade6e8c-ce43-43a9-87ae-9039fabc07ab">

If we click on "Edit Segment" we land again in the same view as before saving the segment:
<img width="750" alt="Screenshot 2024-01-24 at 12 19 45" src="https://github.com/formbricks/formbricks/assets/675065/a0be7413-8960-43f2-b17e-e0714713227d">

Now we are editing the exact Segment (that we may have globally for many different surveys). As soon as we hit the `Save` or `Publish` button of the survey the changes to the segment are saved.

I think it's too easy that people accidentally change a global user segment that is used for multiple surveys and it's not transparent when the save happens.

My suggestion:
- Add an information that currently the `my first segment` segment is being edited.
- Don't save the changes to the segment with the surveys `Save` or `Publish` button
- Add a separate `Save changes` button below the filter editor that saves the changes and let's the view return to the saved filter overview.
<img width="748" alt="Screenshot 2024-01-24 at 12 24 14" src="https://github.com/formbricks/formbricks/assets/675065/053c3121-02da-47ba-add8-5d76216cde78">
- Also give an option to `Cancel` the update/edit process of the segment.

What do you think?